hyperjumptech/monika
# Feat: Configurable Follow HTTP Redirect for Probe Requests

# Monika Pull Request (PR)

## What feature/issue does this PR add

This PR adds the configuration for following redirects on the probe request level. This resolves #1259.

## How did you implement / how did you fix it

- Refactor : Moved global variables to the context to prevent recreation in multiple places.
- Refactor: Extracted logging running information and the function to fetch and cache public network information.
- Refactor: Extracted initialization of Prometheus function.
- Docs: Added documentation for following redirects.
- Feat: Added the follow redirects JSON schema property.
- [Feat: Implemented following redirects on the request level](https://github.com/hyperjumptech/monika/commit/3c1f816b8427606cc98597dae6bf91f4c686cfe6) (The additional feature)
- Fix: Adjusted affected codes

## How to test

1. Run `httpbin` `docker compose -f ./dev/docker-compose.yaml up httpbin -d `.
2. Run Monika with the following configuration. Run `npm start -- -r 1`.

```yaml
probes:
  - id: 'should not follow redirect'
    requests:
      - url: http://localhost:3000/status/302
        followRedirects: 0
    alerts:
      - assertion: response.status != 302
        message: You should not follow redirect
  - id: 'should follow redirect with default config'
    requests:
      - url: http://localhost:3000/absolute-redirect/20
    alerts:
      - assertion: response.status == 302
        message: You are not follow redirect
  - id: 'should follow redirect with customize config'
    requests:
      - url: http://localhost:3000/status/302
        followRedirects: 2
    alerts:
      - assertion: response.status == 302
        message: You are not follow redirect
```

You should see the following output:
<img width="1063" alt="Screenshot 2024-04-16 at 4 50 25 PM" src="https://github.com/hyperjumptech/monika/assets/15191978/9039f3b6-8e58-458a-8d72-ec053dca0c04">
